element-template-generator/core/src/main/java/io/camunda/connector/generator/java/util/TemplatePropertiesUtil.java

```diff
@@ -277,11 +278,29 @@ private static List<PropertyBuilder> handleSealedType(Class<?> type) {
 
       var currentSubTypeProperties =
           extractTemplatePropertiesFromType(subType).stream()
-              .map(
-                  property ->
-                      property.condition(
-                          new PropertyCondition.Equals(
-                              discriminatorIdAndName.getKey(), subTypeIdAndName.getKey())))
+              .peek(
+                  property -> {
+                    if (property.getCondition() == null) {
+                      var condition =
+                          new Equals(discriminatorIdAndName.getKey(), subTypeIdAndName.getKey());
+                      property.condition(condition);
+                    } else {
+                      if (property.getCondition() instanceof AllMatch allMatch) {
+                        var conditions = new ArrayList<>(allMatch.allMatch());
+                        conditions.add(
+                            new Equals(discriminatorIdAndName.getKey(), subTypeIdAndName.getKey()));
+                        property.condition(new AllMatch(conditions));
+                      } else {
+                        property.condition(
+                            new AllMatch(
+                                List.of(
+                                    property.getCondition(),
+                                    new Equals(
+                                        discriminatorIdAndName.getKey(),
+                                        subTypeIdAndName.getKey()))));
+                      }
+                    }
+                  })
```

Comment:
This and the whole method is quite complex. Maybe we can split it up in the future

Reply:
Added a follow-up: https://github.com/camunda/connectors/issues/1565

I will go ahead and merge this now to unblock the SOAP connector activities.